lib/npm-installation-manager.ts

```diff
@@ -130,99 +68,59 @@ export class NpmInstallationManager implements INpmInstallationManager {
 		}).future<string>()();
 	}
 
-	private addCleanCopyToCache(packageName: string, version: string): IFuture<any> {
-		return (() => {
-			let packagePath = path.join(this.getCacheRootPath(), packageName, version);
-			this.$logger.trace(`Deleting: ${packagePath}.`);
-			this.$fs.deleteDirectory(packagePath).wait();
-			let cachedPackageData = this.addToCacheCore(packageName, version).wait();
-			if(!this.isShasumOfPackageCorrect(packageName, cachedPackageData.version).wait()) {
-				this.$errors.failWithoutHelp(`Unable to add package ${packageName} with version ${cachedPackageData.version} to npm cache. Try cleaning your cache and execute the command again.`);
-			}
-
-			return cachedPackageData;
-		}).future<any>()();
-	}
-
-	private addToCacheCore(packageName: string, version: string): IFuture<any> {
+	private installCore(packageName: string, pathToSave: string, version: string, dependencyType: string): IFuture<string> {
 		return (() => {
-			let cachedPackageData = this.$npm.cache(packageName, version).wait();
-			let packagePath = path.join(this.getCacheRootPath(), packageName, cachedPackageData.version, "package");
-			if(!this.isPackageUnpacked(packagePath, packageName).wait()) {
-				this.cacheUnpack(packageName, cachedPackageData.version).wait();
+			// check if the packageName is url or local file and if it is, let npm install deal with the version
+			if(this.isURL(packageName) || this.$fs.exists(packageName).wait()) {
+				version = null;
+			} else {
+				version = version || this.getLatestCompatibleVersion(packageName).wait();
 			}
-			return cachedPackageData;
-		}).future<any>()();
-	}
-
-	private isShasumOfPackageCorrect(packageName: string, version: string): IFuture<boolean> {
-		return ((): boolean => {
-			let shasumProperty = "dist.shasum";
-			let cachedPackagePath = this.getCachedPackagePath(packageName, version);
-			let packageInfo = this.$npm.view(`${packageName}@${version}`, shasumProperty).wait();
 
-			if (_.isEmpty(packageInfo)) {
-				// this package has not been published to npmjs.org yet - perhaps manually added via --framework-path
-				this.$logger.trace(`Checking shasum of package ${packageName}@${version}: skipped because the package was not found in npmjs.org`);
-				return true;
+			if(version === constants.PackageVersion.NEXT) {
```

Comment:
why do you need special code for `next` version. In case we decide to use separate tag (for example `nextNext`) we shouldn't add it here. Npm supports installing of package directly by tag, for example `npm install nativescript@next` will work as expected. Is there a particular reason to have special methods for `next` and `latest` tags?